File: frontend/src/app/core/models/watches/watch-request.dto.ts
import { WatchCondition } from "./enums/watch-condition";
import { WatchGender } from "./enums/watch-gender";
import { WatchMovementType } from "./enums/watch-movement-type";
import { WatchStatus } from "./enums/watch-status";
import { WatchType } from "./enums/watch-type";

export interface WatchRequestDTO {
  manufacturer: string;
  model: string;
  referenceNumber: string;
  serialNumber: string;
  movement: string;
  description: string;
  yearOfProduction: number;
  pricePerDay: number;
  condition: WatchCondition;
  gender: WatchGender;
  movementType: WatchMovementType;
  status: WatchStatus;
  watchType: WatchType;
  branchId: string;
}

File: backend/src/main/java/com/yourwatchrental/watchrental/watch/dto/request/WatchRequestDTO.java
package com.yourwatchrental.watchrental.watch.dto.request;

import com.yourwatchrental.watchrental.watch.enums.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;
import java.util.UUID;

public record WatchRequestDTO(
        @NotBlank(message = "Manufacturer is required")
        String manufacturer,

        @NotBlank(message = "Model is required")
        String model,

        @NotBlank(message = "Reference number is required")
        String referenceNumber,

        @NotBlank(message = "Serial number is required")
        String serialNumber,

        @NotBlank(message = "Movement is required")
        String movement,

        String description,

        @NotNull(message = "Year of production is required")
        @PositiveOrZero(message = "Year of production needs to be accurate")
        int yearOfProduction,

        @NotNull(message = "Price per day is required")
        @PositiveOrZero(message = "Price per day needs to be accurate")
        BigDecimal pricePerDay,

        @NotNull(message = "Condition is required")
        Condition condition,

        @NotNull(message = "Gender is required")
        Gender gender,

        @NotNull(message = "Movement is required")
        MovementType movementType,

        @NotNull(message = "Watch status is required")
        Status status,

        @NotNull(message = "Watch type is required")
        WatchType watchType,

        @NotNull(message = "Branch is required")
        UUID branchId
) {
}
